verilog/analysis/verilog_linter.cc

```diff
@@ -77,57 +89,249 @@ using verible::LintWaiver;
+      absl::string_view path) const override {
+    rule_answers_.clear();
+    answer_ = kMissingAnswer;
+    ViolationFixer::HandleLintRuleStatuses(statuses, base, path);
   }
-  return 0;
-}
+
+ protected:
+  virtual void HandleViolation(const verible::LintViolation& violation,
+                               absl::string_view base, absl::string_view path,
+                               absl::string_view url,
+                               absl::string_view rule_name) const override {
+    assert(fix_ != nullptr);
+
+    formatter_.FormatViolation(stream_, violation, base, path, url, rule_name);
+    (*stream_) << std::endl;
+
+    static absl::string_view help_message =
+        "y - apply fix\n"
+        "n - reject fix\n"
+        "a - apply this and all remaining fixes for violations of this rule\n"
+        "d - reject this and all remaining fixes for violations of this rule\n"
+        "A - apply this and all remaining fixes\n"
+        "D - reject this and all remaining fixes\n"
+        "p - show fix\n"
+        "P - show fixes applied so far\n"
+        "? - print this help and prompt again\n";
+
+    if (violation.autofixes.size() > 0) {
+      char c;
+      for (;;) {
+        if (answer_ != kMissingAnswer) {
+          c = answer_;
+        } else if (rule_answers_.find(rule_name) != rule_answers_.end()) {
+          c = rule_answers_.at(rule_name);
+        } else {
+          c = ReadChar("Autofix is available. Apply? [y,n,a,d,A,D,p,P,?] ");
+        }
+
+        switch (c) {
+          case 'A':
+            answer_ = 'y';
+            [[fallthrough]];
+          case 'a':
+            rule_answers_[rule_name] = 'y';
+            [[fallthrough]];
+          case 'y':
+            if (!fix_->AddEdits(violation.autofixes[0].Edits())) {
+              VLOG(2) << "The fix conflicts with previously applied fixes, "
+                         "rejecting.";
+            }
+            break;
+
+          case '\0':
+            // EOF: received when too few "answers" have been piped to stdin.
+            LOG(WARNING) << "Received EOF while there are questions left. "
+                            "Rejecting all remaining fixes.";
+            [[fallthrough]];
+          case 'D':
+            answer_ = 'n';
+            [[fallthrough]];
+          case 'd':
+            rule_answers_[rule_name] = 'n';
+            [[fallthrough]];
+          case 'n':
+            return;
+
+          case 'p':
+            PrintFix(std::cerr, base, violation.autofixes[0]);
+            continue;
+          case 'P':
+            PrintFix(std::cerr, base, *fix_);
+            continue;
+
+          case 'q':
+            exit(0);  // FIXME: remove
+
+          case '?':
+          default:
+            std::cerr << help_message << std::endl;
+            continue;
+        }
+
+        break;
+      }
+    }
+  }
+
+ private:
+  char ReadChar(absl::string_view prompt) const {
+    if (isatty(0)) std::cerr << prompt;
```

Comment:
`STDIN_FILENO` would be easier to digest for the reader.

Reply:
Done